src/cascade/core/context.py

```diff
@@ -36,51 +37,17 @@ def __iter__(self):
         return iter([self.pini, self.iota, self.rho, self.chi, self.omega])
 
 
-class _Outputs:
-    __slots__ = ["integrands"]
-
-    def __init__(self):
-        self.integrands = _Integrands()
-
-
-class _Integrand:
-    __slots__ = ["name", "age_ranges", "time_ranges", "value_covariate_multipliers", "std_covariate_multipliers"]
+class _IntegrandCovariateMultiplier:
+    __slots__ = ["name", "value_covariate_multipliers", "std_covariate_multipliers"]
 
     def __init__(self, name):
         self.name = name
-        self.age_ranges = None
-        self.time_ranges = None
         self.value_covariate_multipliers = []
         self.std_covariate_multipliers = []
 
 
-class _Integrands:
-    __slots__ = [
-        "Sincidence",
-        "remission",
-        "mtexcess",
-        "mtother",
-        "mtwith",
-        "susceptible",
-        "withC",
-        "prevalence",
-        "Tincidence",
-        "mtspecific",
-        "mtall",
-        "mtstandard",
-        "relrisk",
-    ]
-
-    def __init__(self):
-        for name in self.__slots__:
-            setattr(self, name, _Integrand(name))
-
-    def __iter__(self):
-        for name in self.__slots__:
-            yield getattr(self, name)
-
-
 class ModelContext:
+    __slots__ = ["parameters", "input_data", "rates", "average_integrand_cases", "integrand_covariate_multipliers"]
```

Comment:
I like your carefulness with slots. It avoids attaching things randomly.

Reply:
I do it because I keep getting it wrong and on these objects that exist only to hold attributes, that sucks.